In [15]:
%matplotlib inline
import pandas
import numpy as np

import matplotlib.pyplot as plt
import analysis
reload(analysis)

def convert(i):
    return i*35.7851 - 16.2245

files = ['1_Lead.csv', '2_bronze.csv', '4_5pence.csv', \
         '5_Copper.csv', '6_shinymetal.csv']

peaks = [[(280, 300), (330, 360)], [(215, 230)], [(215, 235)], \
         [(215, 230)], [(170, 185)]]

for index, f in enumerate(files):
    data = pandas.read_csv('data/' + f)
    for peak in peaks[index]:
        a, mean, std = analysis.find_gaussian(data, peak)
        print('{} has a peak at {} +/- {} eV'.format(f, convert(mean), convert(std)))
    

1_Lead.csv has a peak at 10493.3105355 +/- 240.087313859 eV
1_Lead.csv has a peak at 12551.7194963 +/- 306.161772198 eV
2_bronze.csv has a peak at 7993.42433667 +/- 251.55517841 eV
4_5pence.csv has a peak at 8019.72613391 +/- -245.13374874 eV
5_Copper.csv has a peak at 7974.71001325 +/- 207.375066434 eV
6_shinymetal.csv has a peak at 6354.6671517 +/- -171.600722962 eV


Lead peak 1 at 10493 is the Lalpha1 peak L3-M4, 0.7077 probability, and 10551.0 eV.
Lead peak 2 at 12551 is the Lbeta1 peak L2-M4, 0.7878 probability, and 12614.0 eV.

For the copper peaks, we get a mixture of Kalpha2, K-L2, 0.2943 probability, and energy 8026.7eV.  Kalpha1, K-L3, 0.5771 probability, 8046.3 eV.

ShinyMetal peak at 6354 is a mix of iron Kalpha2 which is K-L2 (0.2940 probability), which has an energy of 6392.1 and iron Kalpha1 which is K-L3 with probability (0.5803) and energy of 6405.2 eV 